code/modules/tgui/states/default.dm

```diff
@@ -0,0 +1,58 @@
+ /**
+  * tgui state: default_state
+  *
+  * Checks a number of things -- mostly physical distance for humans and view for robots.
+ **/
+
+GLOBAL_DATUM_INIT(tgui_default_state, /datum/tgui_state/default, new)
+
+/datum/tgui_state/default/can_use_topic(src_object, mob/user)
+	return user.default_can_use_tgui_topic(src_object) // Call the individual mob-overridden procs.
+
+/mob/proc/default_can_use_tgui_topic(src_object)
+	return STATUS_CLOSE // Don't allow interaction by default.
+
+/mob/living/default_can_use_tgui_topic(src_object)
+	. = shared_tgui_interaction(src_object)
+	if(. > STATUS_CLOSE && loc)
+		. = min(., loc.contents_tgui_distance(src_object, src)) // Check the distance...
+	if(. == STATUS_INTERACTIVE) // Non-human living mobs can only look, not touch.
+		return STATUS_UPDATE
+
+/mob/living/carbon/human/default_can_use_tgui_topic(src_object)
+	. = shared_tgui_interaction(src_object)
+	if(. > STATUS_CLOSE)
+		. = min(., shared_living_tgui_distance(src_object)) // Check the distance...
+
+/mob/living/silicon/robot/default_can_use_tgui_topic(src_object)
+	. = shared_tgui_interaction(src_object)
+	if(. <= STATUS_DISABLED)
+		return
+
+	// Robots can interact with anything they can see.
+	var/list/clientviewlist = getviewsize(client.view)
```

Comment:
Range is too big for borgs, use a visibility check here.

Reply:
Done on the next line down